File: raizesequationgrade2/matlab/rgrade2.m
% Metodos numericos     Ano 2017 
% Multiplicação de matrizes
% PUC-Rio Jaime Andrés Castañeda Barbosa

%Programa para calculo das raizes de uma equação do 2 grau

clear all
clc

repetir = true;
while (repetir)
    disp('Entre com os coeficientes:');
    
    A = input('A=');
    B = input('B=');
    C = input('C=');
    
    disc = B^2-4*A*C;
       
    if (A == 0)
       if (B == 0)
           disp('nao existe equacao');
       else
           x1 = -C/B;
           disp('equacao possui uma raiz.');
           disp('raiz=');
           x1
       end
    else
        if (disc < 0.)
            disp('equacao nao possui raiz real');
            disp('');
        elseif (disc == 0.)
            x1 = -B/2*A
            disp('equacao possui somente uma raiz');
            disp('raiz=');
            x1
        else
            x1 = (-B+disc^0.5)/(2*A);
            x2 = (-B-disc^0.5)/(2*A);
            disp('equacao possui duas raizes');
            disp('raizes=');
            x1
            x2
        end
    end
    resposta = input('calcular outra raiz (0-N/1-S)?');
    switch resposta
        case 0 
            repetir = false;
        case 1
            repetir = true;
    end
end

File: raizesequationgrade2/rgrade2.c
/******************************************************************************
Jaime Andres Castañeda Barbosa 
Calculo das raizes de 2 grau

*******************************************************************************/
#include <stdio.h>
#include<math.h>

int main()
{
    double a, b, c;                                         /* coeficientes*/
    double x1, x2;                                          //raizes
    double alpha;
    printf("Entre com os coeficientes (a b c): ");
    scanf("%lf", &a);
    scanf("%lf", &b);
    scanf("%lf", &c);
    if(a==0){
        printf("valor de a não pode ser zero.");            //a esta dividindo a equação quadratica
        return 1;
    }
    alpha = b*b -4*a*c;                                      //valores dentro da raiz
    if(alpha <0){
        printf("raizes imaginarias.");
        
    }
    else if (alpha == 0.0){
        x1 = -b / 2*a;                //a raiz é zero só temos uma divisão
        printf("Uma raiz real : %f", x1);
    }
    else{
        alpha = sqrt(alpha);
        x1 = (-b + alpha) / (2*a);
        x2 = (-b - alpha) / (2*a);
        printf("Duas raizes reais: %f e %f", x1, x2);
    }
    
    return 0;
}
